### PR TITLE
Add Wayland support

### PR DESCRIPTION
**Pull request by vishwamartur**
Related to #87

Add Wayland support to AutoKey.

* Add Wayland support in `lib/autokey/gtkapp.py` and `lib/autokey/qtapp.py` by using `pywayland` and `libinput`.
* Update `lib/autokey/iomediator/keygrabber.py` and `lib/autokey/iomediator/windowgrabber.py` to handle keyboard and mouse events using Wayland-specific methods.
* Update `README.rst` to remove the statement that AutoKey is an X11 application and will not function correctly with Wayland.
* Update `.github/ISSUE_TEMPLATE/bug.yml` to remove the dropdown asking users if they use Xorg or Wayland.
* Add new icons for Wayland support in `lib/autokey/qtui/resources/icons/`.

### CODE REVIEW

**Comment by josephj11** [comment]
Welcome to the AutoKey community, @vishwamartur !

This looks like a major piece of work. Thanks for your contribution!

Here's where we are currently:

Our develop branch has the keyboard and mouse APIs already working in Wayland (but largely untested). The developer who did the work is great when he's here, but tends to be absent for long periods of time - as he is now.

AFAIK, no one has tackled the API calls that use wmctrl on x11 for window manipulation. Does your code address this?

 Our generated API docs are [here](https://autokey.github.io/index.html). They're built from metadata in the code.

(The whole "HighLevel" API is based on the (abandoned) xAutomation project, but it can probably be disabled because it has been broken for some time and no one even noticed it. It would be really useful to have as an enhancement later. No idea if what it does is even possible under Wayland.)

The main problem is that in addition to the developer mentioned above, we have one former lead developer helping tie up loose ends before he leaves entirely, a couple of other developers helping out in small ways, and one junior developer who is very active, but not advanced enough yet to support our project as a main developer.

In short, we currently have no one to evaluate and potentially merge your PR.

Also, our current strategy (being debated) is to merge everything into develop (not main) and then convert that to a beta release when ready which becomes the next main release when tested. Develop is currently a bit out of sync with the main branch, but that is being fixed. The Wayland PR that was merged into develop disabled the x11 functionality for some reason and that hasn't been fixed yet.

Nit: we still need to know if an issue is occurring under x11 or Wayland. The part about not working on Wayland would need to be removed.

You can reply here, but it might be better to post on [Gitter](https://gitter.im/autokey/autokey) which is better for unstructured discussions.

**Comment by Elliria** [comment]
First, I'm not sure if this will be necessary since @sebastiansam55 has been working on Wayland code and it may already be working. It can be tested by cloning AutoKey and testing PR #958 on the **develop** branch.

<mark>**[Wayland testers](https://github.com/autokey/autokey/blob/develop/wayland_install.md) are welcome and feedback is much appreciated!**</mark>

I'm loving how thorough @vishwamartur is in tidying up files like `bug.yml` and `README.rst` as part of making Wayland happen. Perhaps @vishwamartur and @sebastiansam55 could collaborate on the Wayland code?

I'll review this anyway, to the best of my ability, with the up-front warning that (borrowing heavily from Star Wars): "I'm not the one you're looking for." Someone more knowledgeable about Wayland needs to look at this.

That said, this PR shouldn't get put through on the **master** branch and would need to be submitted on the **develop** branch since that's where feature development is done.

It's currently got some issues and I'm not sure how this PR (if it was fixed) and @sebastiansam55's Wayland code on the **develop** branch would interact. With that in mind:

* The change to `bug.yml` seems fine.
* The change to `README.rst` seems fine.
* The change to `debian/control` seems fine, but I'm not familiar with all of the Wayland dependencies.
* The change to `lib/autokey/gtkapp.py` is probably fine. It's just a bunch of import statements. I'm not familiar with the imported libraries/modules, though.
* The change to `lib/autokey/iomediator/keygrabber.py` is probably fine. It's just a bunch of import statements. I'm not familiar with the imported libraries/modules, though.
* The change to `lib/autokey/iomediator/windowgrabber.py` needs a bit of work. The use of upper-case `WAYLAND_AVAILABLE` needs to be made into lower-case `wayland_available` since that's the standard recommended for variables. All of the code down to the `handle_wayland_events` function looks fine to me and won't do any harm. I don't know enough, however, to fully understand what, exactly, the `while True` block of that function does. Someone who knows more about that than me would have to look it over.
* The change to `lib/autokey/qtapp.py` is highly questionable and it's failing **lint (3.11)** with an invalid syntax error on `from pywayland.protocol.way`. It's also failing **pytest (3.11)** because of the incomplete `from pywayland.protocol.way` import statement on line 653 (which is probably the same line that lint is complaining about). I have no idea why all of the existing code gets deleted and is replaced by nothing but a whole bunch of imports, so someone who understands our code and understands how Wayland code works would have to take a look at that.
* I'm not sure why the `.svg` files were provided in the PR since we've already got icons.

Last, but not least, it's failing **test-install (3.11)** because it can't find the `pywayland` module. I believe that issue is in the `lib/autokey/iomediator/keygrabber.py` file.

**Comment by BlueDrink9** [comment]
This is pretty clearly AI-generated rubbish to me. Invalid python syntax and mess, non-existant modules. OP is just spamming repos with bounties available. Closing unless we hear a convincing argument otherwise.
